api/hypershift/v1beta1/hostedcluster_conditions.go

```diff
@@ -195,6 +195,13 @@ const (
 	// This condition is used to track the status of the recovery process and to determine if the HostedCluster
 	// is ready to be used after restoration.
 	HostedClusterRestoredFromBackup ConditionType = "HostedClusterRestoredFromBackup"
+
+	// ControlPlaneToDataPlaneConnectivityHealthy indicates whether the control plane can successfully
+	// reach the data plane components.
+	// When True, control plane has healthy connectivity to data plane nodes.
+	// When False, there are network connectivity issues preventing control plane from reaching the data plane.
+	// A failure here may indicate network policy issues, firewall rules, or infrastructure problems.
+	ControlPlaneToDataPlaneConnectivityHealthy ConditionType = "ControlPlaneToDataPlaneConnectivityHealthy"
```

Comment:
I know naming is hard, and subjective, but this condition name is VERY long. I typically refrain from adding strong
opinions on names, except for user-facing primitives like API, CLI args etc.

In this case I'd suggest to have a more concise and simple name for this condition like `DataPlaneConnectionHealthy`:

1) This implies that it's "ControlPlaneToDataPlane" but much shorter.
2) Connection is a simpler term than "Connectivity".

Reply:
The point that this is only half of the job.
Ths represents ControlPlane -> DataPlane connection

When I'll have more time I'll add another condition to represet DataPlane -> ControlPlane connection.